src/rt/backtrace/dwarf.d

```diff
@@ -71,23 +71,35 @@ int traceHandlerOpApplyImpl(const void*[] callstack, scope int delegate(ref size
     int ret = 0;
     foreach (size_t i; 0 .. callstack.length)
     {
-        char[1536] buffer = void; buffer[0] = 0;
-        char[256] addressBuffer = void; addressBuffer[0] = 0;
+        char[1536] buffer = void;
+        size_t bufferLength = 0;
+
+        void appendToBuffer(Args...)(const(char)* format, Args args)
+        {
+            bufferLength += snprintf(buffer.ptr + bufferLength, buffer.length - bufferLength, format, args);
+            if (bufferLength >= buffer.length)
+                bufferLength = buffer.length - 1;
+        }
 
         if (locations.length > 0 && locations[i].line != -1)
-            snprintf(addressBuffer.ptr, addressBuffer.length, "%.*s:%d ", cast(int) locations[i].file.length, locations[i].file.ptr, locations[i].line);
+        {
+            appendToBuffer("%.*s:%d ", cast(int) locations[i].file.length, locations[i].file.ptr, locations[i].line);
+        }
         else
-            addressBuffer[] = "??:? \0";
+        {
+            buffer[0 .. 5] = "??:? ";
+            bufferLength = 5;
+        }
```

Comment:
Fix 1:
```D
addressBuffer[] = "??:? \0";
```
=> `object.Error@(0): Array lengths don't match for copy` https://run.dlang.io/is/4RQZWh

Reply:
Any chance we can add a test for this to prevent future regressions?

Reply:
It's pretty bad with `-release`, causing a segfault as long as the destination buffer is sufficiently large: https://run.dlang.io/is/NEqZOB. Without `-release` an exception is thrown during stacktrace generation, which is also pretty bad.

The latter is how I stumbled on this. LLVM 6.0 led to DWARF v4 debuginfos, rt.backtrace.dwarf doesn't support that version, so no file/line infos => "??:? \0" + exception (debug druntime only; release druntime compiled with `-release` happily reads beyond the string constant, causing no segfault on x86 apparently). The exception with debug-druntime is swallowed in [my test case](https://github.com/ldc-developers/ldc/blob/1572c91fc0dea24d9aa642c07300b1ca5ba82ea6/tests/codegen/exception_stack_trace.d), it only manifests itself in an incomplete stack trace (single entry).

Reply:
@wilzbach: There's no debug druntime (i.e., compiled with enabled bounds checks) available for CI, right? Then it's untestable.